src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1737,5 +1766,37 @@ enum ReplicationMode
         /// </summary>
         Logical
     }
+
+    /// <summary>
+    /// Specifies server type preference.
+    /// </summary>
+    public enum TargetSessionAttributes : byte
+    {
+        /// <summary>
+        ///  Try to connect to every server in order.
+        /// </summary>
+        Any = 1,
+
+        /// <summary>
+        ///  Secondary (readonly) server is selected for the connection.
+        /// </summary>
+        Secondary = 2,
+
+        /// <summary>
+        /// Secondary (readonly) server is selected for the connection. Falls back to Primary (writable), if none are available.
+        /// </summary>
+        PreferSecondary = 3,
+
+        /// <summary>
+        ///  Primary (writable) server is selected for the connection.
+        /// </summary>
+        Primary = 4,
+
+        /// <summary>
+        /// Primary (writable) server is selected for the connection. Falls back to Secondary (readonly), if none are available.
```

Comment:
Hmm, I toyed with this one myself. I'm really not sure what the use-case is for it though. I guess if it's no-cost to implement, there's no harm in having it.

Reply:
Maybe something like: I would like to read the latest data, but it's ok if it's not up to date.

Reply:
Hmm, yeah I can see a sort of "eventual consistency" thing that you might want there